### ISSUE TITLE
Hanging, crashing, unable to startup across game versions

### ISSUE THREAD

**Issue by TacticalUnicorn** [GTNewHorizons/GT-New-Horizons-Modpack#18236]
### Your GTNH Discord Username

tacticaluni

### Your Pack Version

1.6.1, 1.7.0 RC2, 1.7.0 RC3

### Your Server

Private Server

### Java Version

Java 8

### Type of Server

Vanilla Forge

### Your Actions

I have spent the last 3 days troubleshooting for many hours. Java 17 worked fine for 1.6.1, however the server host I am using can only host java 8 since they don't support launch args (*grumble*). I have 2000 hours on the same installation and took a break for a few months. Every time I launched, the game would either crash our my entire PC would hang more often. The error kept changing, but often times it was an illegal access error.
I moved on to 1.7.0 RC2 and the game was working fine for a few hours. I logged off for a bit and came back, and now the game is crashing again. It threw errors about super tic, thaumic horizons and now I'm getting illegal access errors from gregtech.api.util.GTShapedRecipe. My system is also showing 16/32 GB ram usage, but resource monitor and task manager are not showing any processes that are consuming much ram. RC3 would not launch out right. I will happily donate to the server if someone can solve this issue, im losing my mind LOL. 

### Crash Report

[hs_err_pid28312.log](https://github.com/user-attachments/files/18035314/hs_err_pid28312.log)
[fml-client-latest.log](https://github.com/user-attachments/files/18035321/fml-client-latest.log)
[fml-client-3.log](https://github.com/user-attachments/files/18035322/fml-client-3.log)
[fml-client-2.log](https://github.com/user-attachments/files/18035323/fml-client-2.log)
[fml-client-1.log](https://github.com/user-attachments/files/18035324/fml-client-1.log)
[2024-12-05-2.log.gz](https://github.com/user-attachments/files/18035325/2024-12-05-2.log.gz)
[2024-12-05-3.log.gz](https://github.com/user-attachments/files/18035326/2024-12-05-3.log.gz)
[2024-12-05-4.log.gz](https://github.com/user-attachments/files/18035327/2024-12-05-4.log.gz)
[2024-12-06-1.log.gz](https://github.com/user-attachments/files/18035328/2024-12-06-1.log.gz)
[2024-12-06-2.log.gz](https://github.com/user-attachments/files/18035329/2024-12-06-2.log.gz)
[2024-12-05-1.log.gz](https://github.com/user-attachments/files/18035330/2024-12-05-1.log.gz)

### Final Checklist

- [X] I have searched this issue tracker and there is nothing similar already. Posting on a closed issue saying the crash still exists will prompt us to investigate and reopen it once we confirm your report.
- [X] I can reproduce this crash consistently by following the exact steps I described above.
- [ ] I have asked other people and they confirm they also crash by following the exact steps I described above.

**Comment by TacticalUnicorn** [comment]
Troubleshooting steps I've undergone:

- Updating graphic drivers
- Fresh java installation
- Updating dependencies
- Updating version
- Praying (I'm not religious one bit)
- Switching launchers
- Uninstalling possible problematic programs
- Fresh mod pack installations
- Disabling all sound devices
- Forcing regeneration of bugged config files
- Banging my head against the wall as hard as I can
- Removing all programs from startup

**Comment by boubou19** [comment]
test your ram sticks with memtest and see if it has errors.

**Comment by boubou19** [comment]
then something else on your hardware is doing funky things, as you should be able to at least boot 2.6.1

**Comment by TacticalUnicorn** [comment]
I understand. I'm just having difficulty understanding the root of the issue. The pack was working for all of a day before it started throwing errors again. This is also the only pack my computer is having issues with; I've tried 2 other packs and I'm certainly not having any issues with anything else. I've also run as many system diagnostic tools as I'm aware of.

**Comment by TacticalUnicorn** [comment]
I've remedied the issue. I've changed server providers to one who supports JVM arguments so I can run Java 17. As I stated previously the java 17 version is running fine.

**Comment by SKProCH** [comment]
You can use different java version on the server and the client, its doesn't matter.

**Comment by Glease** [comment]
tl;dr: try add `-Djava.util.concurrent.ForkJoinPool.common.parallelism=2` as  jvm argument. This will significantly slow down pack boot, but hopefully it won't crash any more. you can tune up the number here step by step until it starts crashing

If your server or client crash with BSOD or segfault during gt materials baking, then you have a faulty CPU/ram. We parallelize super hard here to get the pack booted faster, but it does place unusual stress on your hardware. We have player using 13900ks to experience the same before. He got it fixed by disabling some CPU cores and then later with the jvm args. Later CPU stress test proved his cpu was hit by that infamous intel defect
